amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/fanout/FanOutRecordsPublisher.java

```diff
@@ -511,8 +512,19 @@ private void recordsReceived(RecordFlow triggeringFlow, SubscribeToShardEvent re
     }
 
     private boolean isValidEvent(SubscribeToShardEvent event) {
```

Comment:
Still not a fan of duplicating same validation code for S2SEvent and GetRecords. We can have a common util or static method like `isValidEvent(KCLEvent event)` and define KCLEvent as
```
@Value
class KCLEvent {
String continuationToken;
List<ChildShard> childshards;
}
```

Reply:
Now put the validation to a utility class with a method that takes in a string as the continuationSequenceNumber or nextShardIterator, and a list of childShards